### PR TITLE
#9 - Create a box component

### LINKED ISSUE #9: Create a box component
- Behaves like a col by default
- Reuse the existing Row/Col
- Support spacing types such as tiny, medium, large, xLarge etc. by default. Also custom values can be passed.

// When true, direction = row
row?: boolean

/**
   * Border radius facilitates curved borders
   */
  borderRadius?: number;

  /**
   * Children components.
   */
  children?: React.ReactNode;

  /**
   * Background color for the row. Useful in dev environment.
   */
  color?: Color;

  /**
   * Horizontally center align child components
   */
  hCenter?: boolean;

  /**
   * Bottom align child components
   */
  bottom?: boolean;

  /**
   * Vertically center align child components
   */
  vCenter?: boolean;

  /**
   * Height of the container
   */
  height?: number | string;

  /**
   * Width of the container
   */
  width?: number | string;

  /**
   * Maximum height of the container
   */
  maxHeight?: number | string;

  /**
   * Maximum width of the container
   */
  maxWidth?: number | string;

  /**
   * Flex size
   */
  flex?: number;

  /**
   * Flex wrap
   */
  wrap?: boolean;

  /**
   * Hides the view when set to true
   */
  hidden?: boolean;

  /**
   * Component margin top
   */
  marginTop?: SpacingTypes;

  /**
   * Component margin bottom
   */
  marginBottom?: SpacingTypes;

  /**
   * Component vertical padding
   */
  paddingVertical?: SpacingTypes;

  /**
   * Component horizontal padding
   */
  paddingHorizontal?: SpacingTypes;

  /**
   * Component vertical margin
   */
  marginVertical?: SpacingTypes;

  /**
   * Component horizontal margin
   */
  marginHorizontal?: SpacingTypes;

  /**
   * Component right margin
   */
  marginRight?: SpacingTypes;

  /**
   * Component padding
   */
  padding?: SpacingTypes;

  /**
   * Component margin
   */
  margin?: SpacingTypes;

  /**
   * Aligns content to right
   */
  rightAlign?: boolean;

  /**
   * Opacity pf the grid
   */
  opacity?: number;

  /**
   * An optional style override useful for padding & margin.
   */
  style?: ViewStyle;

### PR DESCRIPTION
**Pull request by chanchal-sadhukhan**
Description: 
 - Box component added

### CODE REVIEW

File: src/components/primitive/box/box.component.props.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+  /**
+    Vertically center align child components
+    */
+  vCenter?: boolean;
+
+  /**
+    Height of the container
+    */
+  height?: number | string;
```
Comment (dsouzaedison) [inline src/components/primitive/box/box.component.props.ts]: Height/width can be a string?

**Review by dsouzaedison — CHANGES_REQUESTED** [review]

File: src/components/primitive/box/box.component.props.ts (hunk 1 above)
Comment (koshikraj) [reply src/components/primitive/box/box.component.props.ts]: It makes sense if we are passing the type along with it (Ex: `10px`, `1rem`). I think we haven't defined any standard type for this.

Comment (chanchal-sadhukhan) [reply src/components/primitive/box/box.component.props.ts]: we are using rem

**Review by dsouzaedison — APPROVED** [review]

**Comment by dsouzaedison** [comment]
Wait for @Koushith to merge his PR's (#7 and #10 )

**Comment by dsouzaedison** [comment]
Update this PR with theming changes by taking the latest from develop.

**Comment by chanchal-sadhukhan** [comment]
I'll update the theming

**Pull request merged**